stacks/api/bucket/hooks/src/queue.ts

```diff
@@ -0,0 +1,48 @@
+import {Action} from "@spica-server/bucket/hooks/proto";
+import {Queue} from "@spica-server/function/queue";
+import * as grpc from "grpc";
+
+export class ActionQueue implements Queue<typeof Action.Queue> {
+  TYPE = Action.Queue;
+
+  callbacks = new Map<string, Function>();
+  queue = new Map<string, Action.Action>();
+
+  get size() {
+    return this.queue.size;
+  }
+
+  pop(call: grpc.ServerUnaryCall<Action.Action.Pop>, callback: grpc.sendUnaryData<Action.Action>) {
+    const action = this.queue.get(call.request.id);
+    if (!this.queue.has(call.request.id)) {
+      return callback(new Error(`Queue has no item with id ${call.request.id}`), null);
+    }
+    this.queue.delete(call.request.id);
+    callback(null, action);
+  }
+
+  result(
+    call: grpc.ServerUnaryCall<Action.Result>,
+    callback: grpc.sendUnaryData<Action.Result.Response>
+  ) {
+    const _callback = this.callbacks.get(call.request.id);
+    if (!this.callbacks.has(call.request.id)) {
+      return callback(new Error(`Queue has no callback with id ${call.request.id}`), null);
+    }
+    this.queue.delete(call.request.id);
```

Comment:
This line was written by me. 👯‍♂  Shall we remove the callback from the callbacks map instead?

In this case, this line should be something like `this. callbacks.delete(call.request.id);`